backend/lib/src/data/storage/memory.rs

```diff
@@ -21,21 +32,75 @@ pub enum InMemoryStorageError {
     // This enum is kept for future extensibility
 }
 
-/// In-memory storage implementation using Arc<RwLock<HashMap>>
-///
-/// TODO(SCAFFOLDING): Example storage implementation for demonstration.
-/// Replace with actual persistent storage in production.
+/// Nonce entry with address and expiration timestamp
+#[derive(Clone, Debug)]
+struct NonceEntry {
+    address: String,
+    expires_at: Instant,
+}
+
+/// In-memory storage implementation
 ///
 /// This implementation is thread-safe and suitable for development environments.
 /// All data is lost when the process terminates.
-#[derive(Default, Clone)]
+#[derive(Clone)]
 pub struct InMemoryStorage {
```

Comment:
Documents the fields. Are you storing nonces? Docs.
Also, why this InMemoryStorage needs to implement a trait and not directly impl the struct?

Reply:
> Also, why this InMemoryStorage needs to implement a trait and not directly impl the struct?

We could move this "storage" into something else, like the DB has mock/postgres, this one could be mock (in memory)/ redis